### DISCUSSION TITLE
Lateset Update All seems to have broken the Manager

### DISCUSSION THREAD

**Discussion by EddieSoloProjects** [Comfy-Org/ComfyUI-Manager#235] [Q&A]
I asked here as well: https://github.com/comfyanonymous/ComfyUI/discussions/2262#discussioncomment-7825454
After Updating All , Now the Manager seems to be broken and can no longer update or install anything.

**Comment by ltdrdata** [comment]
So I recommended the command line way. `git pull`.

**Comment by EddieSoloProjects** [comment]
Thanks That worked :)

**Comment by EddieSoloProjects** [comment]
Actually I spoke too soon. Manager now opens without an error but when you try an update you get "ComfyUI update fail: name 'git' is not defined" Error

**Reply by ltdrdata** [reply]
It seems your installation environment isn't valid. 
What is your ComfyUI envirnoment?

OS? 
portable version?
venv?

**Reply by EddieSoloProjects** [reply]
I just downloaded the zip with the portable and ran it that way. It's been working for 1 week. Today I did an Update All and when I relaunched it it was broken. Can I set the Environment manually in some file?

**Reply by ltdrdata** [reply]
Update ComfyUI-Manager again. And send me the startup log.

You can find path of startup log on startup message.

```
** ComfyUI startup time: 2023-12-12 12:39:23.460983
** Platform: Linux
** Python version: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
** Python executable: /home/me/github/ComfyUI/venv/bin/python
** Log path: /home/me/github/ComfyUI/comfyui.log
```

**Reply by EddieSoloProjects** [reply]
### Loading: ComfyUI-Manager (V1.11.2)
## ComfyUI-Manager: installing dependencies
 Requirement already satisfied: GitPython in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from -r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 1)) (3.1.40)
 Requirement already satisfied: matrix-client==0.4.0 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from -r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (0.4.0)
 Requirement already satisfied: requests~=2.22 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (2.31.0)
 Requirement already satisfied: urllib3~=1.21 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (1.26.18)
 Requirement already satisfied: gitdb<5,>=4.0.1 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from GitPython->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 1)) (4.0.11)
 Requirement already satisfied: smmap<6,>=3.0.1 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from gitdb<5,>=4.0.1->GitPython->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 1)) (5.0.1)
 Requirement already satisfied: charset-normalizer<4,>=2 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from requests~=2.22->matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (3.3.2)
 Requirement already satisfied: idna<4,>=2.5 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from requests~=2.22->matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (3.4)
 Requirement already satisfied: certifi>=2017.4.17 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from requests~=2.22->matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (2023.7.22)
## [ERROR] ComfyUI-Manager: Attempting to reinstall dependencies using an alternative method.
 Requirement already satisfied: GitPython in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from -r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 1)) (3.1.40)
 Requirement already satisfied: matrix-client==0.4.0 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from -r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (0.4.0)
 Requirement already satisfied: requests~=2.22 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (2.31.0)
 Requirement already satisfied: urllib3~=1.21 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (1.26.18)
 Requirement already satisfied: gitdb<5,>=4.0.1 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from GitPython->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 1)) (4.0.11)
 Requirement already satisfied: smmap<6,>=3.0.1 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from gitdb<5,>=4.0.1->GitPython->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 1)) (5.0.1)
 Requirement already satisfied: charset-normalizer<4,>=2 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from requests~=2.22->matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (3.3.2)
 Requirement already satisfied: idna<4,>=2.5 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from requests~=2.22->matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (3.4)
 Requirement already satisfied: certifi>=2017.4.17 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from requests~=2.22->matrix-client==0.4.0->-r D:\AI_COMFY\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\requirements.txt (line 2)) (2023.7.22)
## [ERROR] ComfyUI-Manager: Failed to install the GitPython package in the correct Python environment. Please install it manually in the appropriate environment. (You can seek help at https://app.element.io/#/room/%23comfyui_space%3Amatrix.org)
## ComfyUI-Manager: installing dependencies done.
### ComfyUI Revision: UNKNOWN (The currently installed ComfyUI is not a Git repository)

**Reply by ltdrdata** [reply]
goto cmd and execute this command.

```
d:\ai_comfy\comfyui_windows_portable\python_embeded\python.exe -m pip install GitPython
```

What message is displayed?

**Reply by EddieSoloProjects** [reply]
C:\WINDOWS\system32>d:\ai_comfy\comfyui_windows_portable\python_embeded\python.exe -m pip install GitPython
Requirement already satisfied: GitPython in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (3.1.40)
Requirement already satisfied: gitdb<5,>=4.0.1 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from GitPython) (4.0.11)
Requirement already satisfied: smmap<6,>=3.0.1 in d:\ai_comfy\comfyui_windows_portable\python_embeded\lib\site-packages (from gitdb<5,>=4.0.1->GitPython) (5.0.1)

**Reply by ltdrdata** [reply]
And try this

`d:\ai_comfy\comfyui_windows_portable\python_embeded\python.exe -c "import git"`

**Reply by EddieSoloProjects** [reply]
C:\WINDOWS\system32>d:\ai_comfy\comfyui_windows_portable\python_embeded\python.exe -c "import git"
Traceback (most recent call last):
  File "d:\ai_comfy\comfyui_windows_portable\python_embeded\Lib\site-packages\git\__init__.py", line 166, in <module>
    refresh()
  File "d:\ai_comfy\comfyui_windows_portable\python_embeded\Lib\site-packages\git\__init__.py", line 153, in refresh
    if not Git.refresh(path=path):
           ^^^^^^^^^^^^^^^^^^^^^^
  File "d:\ai_comfy\comfyui_windows_portable\python_embeded\Lib\site-packages\git\cmd.py", line 391, in refresh
    raise ImportError(err)
ImportError: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "d:\ai_comfy\comfyui_windows_portable\python_embeded\Lib\site-packages\git\__init__.py", line 168, in <module>
    raise ImportError("Failed to initialize: {0}".format(_exc)) from _exc
ImportError: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

**Reply by ltdrdata** [reply]
You must not installed `git` or it is outdated.

https://github.com/ltdrdata/ComfyUI-Manager?tab=readme-ov-file#installationmethod1-general-installation-method-comfyui-manager-only

**Reply by EddieSoloProjects** [reply]
Git was installed on 11/22/2023 same with Git LFS

**Reply by EddieSoloProjects** [reply]
It's the latest build for git for windows

**Reply by EddieSoloProjects** [reply]
I Re-installed it and now the Manager is working :)

**Reply by EddieSoloProjects** [reply]
Thanks !!! Comfy is NO FUN without this Plugin !!
